pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/CaseTransformFunction.java

```diff
@@ -104,147 +104,62 @@ private TransformResultMetadata calculateResultMetadata() {
       Preconditions.checkState(thenStatementResultMetadata.isSingleValue(),
           String.format("Unsupported multi-value expression in the THEN clause of index: %d", i));
       DataType thenStatementDataType = thenStatementResultMetadata.getDataType();
+      if (thenStatementDataType == dataType) {
```

Comment:
nit: perhaps add a comment for this method about how type is determined. iiuc, it seems like we try to find the most common type for all `then clauses` and `else clause` that can be cast to.

Reply:
Good point, added